rpc/handlers.go

```diff
@@ -1067,15 +1067,15 @@ func (h *Handler) AddTransaction(txnJSON json.RawMessage) (*AddTxResponse, *json
 		return nil, jsonrpc.Err(jsonrpc.InvalidJSON, err.Error())
 	}
 
-	if txnType, typeFound := request["type"]; typeFound && txnType == TxnInvoke.String() {
+	if txnType := request["type"]; txnType == TxnInvoke.String() {
 		request["type"] = starknet.TxnInvoke.String()
 
 		updatedReq, errIn := json.Marshal(request)
 		if errIn != nil {
 			return nil, jsonrpc.Err(jsonrpc.InternalError, errIn.Error())
 		}
 		txnJSON = updatedReq
-	} else if version, ok := request["version"]; ok && (version == "0x2" || version == "0x3") {
+	} else if version := request["version"]; txnType == TxnDeclare.String() && (version == "0x2" || version == "0x3") {
 		contractClass, ok := request["contract_class"].(map[string]interface{})
```

Comment:
I'm also wondering, why part of validation we have like above, and other like that? 
https://github.com/NethermindEth/juno/blob/9448c9f56c33196c8aacc1d86c3cd23d3d7b424b/rpc/transaction.go#L215

Reply:
No particular reason, it is tech debt that we will have to address at some point. https://github.com/NethermindEth/juno/issues/1477